### PR TITLE
SDK-124 - Develop Unit Tests for service package

### PR DESCRIPTION
**Pull request by AmHubert256**

### CODE REVIEW

File: symphony-bdk-legacy/symphony-api-client-java/src/test/java/services/AbstractDatafeedEventsServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+        assertEquals(initialSize - 1, newSize);
+
+        service.removeListeners(imListener3);
+        newSize = service.getIMListener().size();
+        assertEquals(initialSize - 2, newSize);
+    }
+
+    // Anonymous class creation
+    private RoomListener createRoomListener() {
+        return new RoomListener() {
```
Comment (symphony-thibault) [inline symphony-bdk-legacy/symphony-api-client-java/src/test/java/services/AbstractDatafeedEventsServiceTest.java:171]: Could we use Mockito to get a new instance for this interface? 

```java
private RoomListener createRoomListener() {
    return mock(RoomListener.class);
}
```

File: symphony-bdk-legacy/symphony-api-client-java/src/test/java/services/AbstractDatafeedEventsServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+            public void onRoomUpdated(RoomUpdated roomUpdated) {}
+            @Override
+            public void onUserJoinedRoom(UserJoinedRoom userJoinedRoom) {}
+            @Override
+            public void onUserLeftRoom(UserLeftRoom userLeftRoom) {}
+        };
+    }
+
+    private ConnectionListener createConnectionListener() {
+        return new ConnectionListener() {
```
Comment (symphony-thibault) [inline symphony-bdk-legacy/symphony-api-client-java/src/test/java/services/AbstractDatafeedEventsServiceTest.java:194]: Same comment as above, could we use `mock(ConnectionListener.class)` to generate new instances?

File: symphony-bdk-legacy/symphony-api-client-java/src/test/java/services/DatafeedEventsServiceV1Test.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        when(datafeedClient.createDatafeed()).thenReturn("123");
+        when(datafeedClient.getAckId()).thenReturn("ack_id_string");
+    }
+
+    @SneakyThrows
+    @Test
+    public void readDatafeedTest() {
+        datafeedEventsService = new DatafeedEventsService(symBotClient);
+        // Wait for async operation to finish
+        Thread.sleep(1500);
```
Comment (symphony-thibault) [inline symphony-bdk-legacy/symphony-api-client-java/src/test/java/services/DatafeedEventsServiceV1Test.java:39]: Is it possible to reduce the sleep duration? `1500` will impact our global tests execution time, what could be the minimal value?

Comment (AmHubert256) [reply symphony-bdk-legacy/symphony-api-client-java/src/test/java/services/DatafeedEventsServiceV1Test.java]: I reduced the duration to `500`

**Pull request merged**